### PR TITLE
Fix lasso unresponsive issue by adding a lock release event

### LINKED ISSUE #25137: [Bug]: stop responding in demo program "matplotlib/examples/event_handling/lasso_demo.py"
### Bug summary

The demo shows some points. You may surround some of them with a "lasso" (pressing the left mouse button an moving the cursor arround) and then these points are displayed in a different colour. Trying another lasso, other points will be coloured and so on. 
But if you just click the left mouse button once and DO NOT MOVE THE CURSOR, then the program hangs and no other lassos can be thrown.

### Code for reproduction

```python
just matplotlib/examples/event_handling/lasso_demo.py
just try
```

### Actual outcome

none, as described in bug summary

### Expected outcome

The program should ignore such a behaviour (only click one + no move) and then should produce a correct set of coloured points on a further lasso example.

### Additional information

The function "lasso" does not recognize, that NO movement happens.
But in the demo file after EVERY lasso call a 
        **# acquire a lock on the widget drawing
        self.canvas.widgetlock(self.lasso)**
is executed, so that the program is unable to receive another "lasso" if the error mentioned above appears.

### Operating system

win 10

### Matplotlib Version

3.5.1

### Matplotlib Backend

TkAgg

### Python version

3.8.10

### Jupyter version

_No response_

### Installation

pip

### PR DESCRIPTION
**Pull request by devRD**
## PR Summary
Fixes #25137 by adding a lock release event. 

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

**Comment by jklymak** [comment]
@ianhi maybe you are the wrong person to ask for a review here, but feel free to suggest someone else...

File: examples/event_handling/lasso_demo.py
Comment (ianhi) [inline examples/event_handling/lasso_demo.py:92]: You need to also check that the lasso is the current owner otherwise you risk throwing an error https://github.com/matplotlib/matplotlib/blob/39070aeaab732eca1b5b7f7b478fa8b72dab8507/lib/matplotlib/widgets.py#L50:

```suggestion
        if self.canvas.widgetlock.isowner(self.lasso):
            self.canvas.widgetlock.release(self.lasso)
```

**Review by ianhi — APPROVED** [review]
I'd approve this (conditional on my comment about checking for ownership of the lock) as this does indeed fix the reported bug. Nice fix @devRD!

That said I think that using `Lasso` is something that should be discouraged in favor of lasso selector - perhaps even deprecated and removed. See my comments in https://github.com/matplotlib/matplotlib/issues/19221 for more. Still this is clear improvement and the perfect should not be the enemy of the good. 

> @ianhi maybe you are the wrong person to ask for a review here,

I'm probably the right person :)

**Comment by devRD** [comment]
@ianhi Thank you for the review! I added your suggested changes.

**Comment by ianhi** [comment]
Awesome @devRD - i think that the process is now that at least one core dev (i.e. not me) needs to approve this (only one as this is documentation).

For whoever that is I think the things to check are:

1. Run the example - click once, then see that you can still lasso as expected
2. Run the example make lassos as expected


full file for copy paste purposes: https://github.com/matplotlib/matplotlib/blob/418c9e7037896bea4a4db5cce656f2933014f0e0/examples/event_handling/lasso_demo.py

**Comment by jklymak** [comment]
@ianhi you have merge rights, if I'm not mistaken.

**Comment by ianhi** [comment]
> @ianhi you have merge rights, if I'm not mistaken.

I actually don't - I think I only have the power to open/close issues+PRs and apply labels. I have merging rights over on ipympl

File: examples/event_handling/lasso_demo.py
Comment (QuLogic) [inline examples/event_handling/lasso_demo.py:91]: If the widget lock is already taken in `on_press`, then it would return and not set `self.lasso`. So you cannot use `self.lasso` here directly or you will get an `AttributeError`. This needs to check
```suggestion
        if hasattr(self, 'lasso') and self.canvas.widgetlock.isowner(self.lasso):
            self.canvas.widgetlock.release(self.lasso)
```

**Review by QuLogic — APPROVED** [review]

**Comment by QuLogic** [comment]
Thanks @devRD! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**
